pkg/controller/tiers/tiers_controller.go

```diff
@@ -136,7 +145,48 @@ func (r *ReconcileTiers) Reconcile(ctx context.Context, request reconcile.Reques
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
-	component := tiers.Tiers(&tiers.Config{Openshift: r.provider == operatorv1.ProviderOpenShift})
+	tiersConfig := tiers.Config{
+		Openshift:    r.provider == operatorv1.ProviderOpenShift,
+		NodeLocalDNS: false,
+		KubeDNSCIDR:  "0.0.0.0/32",
+	}
+
+	if r.provider != operatorv1.ProviderOpenShift {
+		nodeLocalDNSExists, err := utils.IsNodeLocalDNSAvailable(ctx, r.client)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying node-local-dns pods", err, reqLogger)
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		} else if nodeLocalDNSExists {
+			// discover kube-dns ip address - node-local-dns is not supported on OpenShift which is the only platform without
+			// kube-dns. Thus, the name "kube-dns" can be static.
+			kubeDNSService := &corev1.Service{}
+			err = r.client.Get(ctx, types.NamespacedName{Name: "kube-dns", Namespace: "kube-system"}, kubeDNSService)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					r.status.SetDegraded(operatorv1.ResourceNotFound, "kube-dns service not found", err, reqLogger)
+				} else {
+					r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying kube-dns service", err, reqLogger)
+				}
+				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+			kubeDNSIpAddress := kubeDNSService.Spec.ClusterIP
```

Comment:
Looks like you're handling IPv6 already, which is great. However, we also need to handle dual-stack services here as well. Instead of looking at `Spec.ClusterIP`, we will need to look at `Spec.ClusterIPs` which can contain up to two IP addresses.

Reply:
ref: https://kubernetes.io/docs/concepts/services-networking/dual-stack/